OSPSuite.Presentation.Importer/Infrastructure/ExcelReader.cs

```diff
@@ -0,0 +1,165 @@
+            _book = WorkbookFactory.Create(fs);
+         }
+
+         _sheetEnumerator = _book.GetEnumerator();
+         _columnOffsetOn = columnOffsetOn;
+         _isTypeXlsx = Path.GetExtension(path).Equals(".xlsx");
+      }
+
+      public bool MoveToNextSheet()
+      {
+         if (!_sheetEnumerator.MoveNext())
+            return false;
+
+         CurrentSheet = _sheetEnumerator.Current;
+
+         _rowEnumerator = CurrentSheet?.GetEnumerator();
+         determineFirstColumn();
+
+         return true;
+      }
+
+      public bool MoveToNextRow()
+      {
+         if (!_rowEnumerator.MoveNext())
+            return false;
+
+         CurrentRow = new List<string>();
+
+         var currentExcelRow = getCurrentExcelRow(_rowEnumerator);
+
+         for (var i = _columnOffset; i < currentExcelRow.LastCellNum; i++)
+         {
+            var cell = currentExcelRow.GetCell(i);
+
+            CurrentRow.Add(cell == null ? "" : cell.ToString());
+         }
+
+         return true;
+      }
+
+      private bool isNumeric(ICell cell)
+      {
+         if (cell.CellType == CellType.Numeric)
+            return true;
+         
+         var value = cell.ToString().TrimStart();
+         if (value.IndexOf('<') == 0)
+            value = value.Substring(1);
+
+         return double.TryParse(value, out _);
+      }
+
+      public List<ColumnDescription.MeasurementLevel> GetMeasurementLevels(int rowLength) //IMPORTANT: should be called after headers have been read!!!!
+      {
+         var resultList = new List<ColumnDescription.MeasurementLevel>();
+         var currentExcelRow = getCurrentExcelRow(_rowEnumerator);
+
+         for (var i = _columnOffset; i < rowLength + _columnOffset; i++)
+         {
+            var cell = currentExcelRow.GetCell(i);
+
+            if (cell == null)
+               resultList.Add(ColumnDescription.MeasurementLevel.NotSet);
+            else if (isNumeric(cell))
+               resultList.Add(ColumnDescription.MeasurementLevel.Numeric);
+            else
+               resultList.Add(ColumnDescription.MeasurementLevel.Discrete);
+         }
+
+         return resultList;
+      }
+
+      private void determineFirstColumn()
+      {
+         if (!_columnOffsetOn)
+         {
+            _columnOffset = 0;
+            return;
+         }
+
+         var excelRows = CurrentSheet.GetRowEnumerator();
+
+         if (!excelRows.MoveNext())
+         {
+            _columnOffset = 0;
+            return;
+         }
+
+         var currentExcelRow = getCurrentExcelRow(excelRows);
+
+         var tableStart = 0;
+
+         for (var i = 0; i < currentExcelRow.LastCellNum; i++)
+         {
+            var cell = currentExcelRow.GetCell(i);
+
+            if (cell == null)
+               tableStart++;
+            else
+               break;
+         }
+
+         _columnOffset = tableStart;
+      }
+
+      private IRow getCurrentExcelRow(IEnumerator enumerator)
+      {
+         IRow row;
```

Comment:
simplify this. 
RETurn right away instead of allocating and returning

Reply:
done